packages/core/src/plugins/default/markbind-plugin-plantuml.ts

```diff
@@ -16,28 +16,35 @@ import { PluginContext } from '../Plugin';
 import { NodeProcessorConfig } from '../../html/NodeProcessor';
 import { MbNode } from '../../utils/node';
 
+const LockManager = require('../../utils/LockManager');
+
 const JAR_PATH = path.resolve(__dirname, 'plantuml.jar');
 
 const processedDiagrams = new Set();
 
 let graphvizCheckCompleted = false;
+// Generate a lock file
 
 /**
  * Generates diagram and returns the file name of the diagram
  * @param imageOutputPath output path of the diagram to be generated
  * @param content puml dsl used to generate the puml diagram
  */
 function generateDiagram(imageOutputPath: string, content: string) {
+  const lockId = LockManager.createLock();
```

Comment:
would it work if we move this to before line 48? to avoid the need for the change at line 38?

Reply:
Yes it will work.